### PR TITLE
Bst

### PR DESCRIPTION
**Pull request by SammyRobensParadise**

### CODE REVIEW

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee  if you could take a loot at this that would be great, I think I am still missing someting but am not sure quite what

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee  still not working.. I think It could be why remove isnt working either.. fuck...

**Comment by SammyRobensParadise** [comment]
ill ping you when you should take a look

**Comment by SammyRobensParadise** [comment]
tl:dr, remove is almost done, the parts that still need to be written are the swap cases, (case 3 in the notes) @maryrsmcphee  lemme know if you think of a way to approach this

**Comment by maryrsmcphee** [comment]
Sorry yes go ahead and merge it and I can look at it in the morning?

**Review by maryrsmcphee — APPROVED** [review]

File: Lab3/binary_search_tree.cpp [hunk 1]
```diff
@@ -187,88 +188,95 @@ bool BinarySearchTree::insert(BinarySearchTree::TaskItem val) {
...
+        // go left
+        return insert(val, node->left);
     }
 }
 
 // PURPOSE: Removes the node with the value val from the tree
 // returns true if successful; returns false otherwise
 bool BinarySearchTree::remove(BinarySearchTree::TaskItem val) {
-    if (root == NULL) {
+    if (root == NULL || &val == NULL) {
```
Comment (maryrsmcphee) [inline Lab3/binary_search_tree.cpp:228]: `&val == NULL` will always be false

Comment (SammyRobensParadise) [reply Lab3/binary_search_tree.cpp:228]: true yea okay

**Pull request merged**